docs/specifications/tests/Basic-TP/basic02.md

```diff
@@ -1,45 +1,77 @@
-## BASIC02: The domain must have at least one working name server
+# BASIC02: The domain must have at least one working name server
 
-### Test case identifier
+## Test case identifier
 **BASIC02** The domain must have at least one working name server
 
-### Objective
+## Objective
 
 In order for the domain to work, it must have at least one name server that
 can answer queries about the domain. 
 
-### Inputs
+## Inputs
+
+* The label of the domain name to be tested ("child zone").
+* If undelegated test, the list of name servers for _the child zone_ 
+  ("undelegated NS").
+* If undelegated test, any IP addresses of the [in-bailiwick] 
+  _undelegated NS_ ("undelegated glue IP").
+* If undelegated test, any IP addresses of the [out-of-bailiwick]
+  _undelegated NS_ ("undelegated non-glue IP").
+
+## Ordered description of steps to be taken to execute the test case
+
+1. Retrieve the NS records for _the child zone_ (the delegation) using
+   [Method 2]. If the test is an undelegated test, use _undelegated
+   NS_.
+2. If no NS records could be retrieved this test case fails
+   ("domain name does not exist").
+3. Retrieve the IP addresses (glue records) for any [in-bailiwick] name
+   servers using [Method 4]. If the test is an undelegated test, use 
+   _undelegated glue IP_.
+4. Retrieve the IP addresses for any [out-of-bailiwick] name servers
+   using recursive queries. If the test is an undelegated test, use 
+   _undelegated non-glue IP_ all such name servers that have IP 
+   address data and do a recursive lookup for the rest.
+5. If no IP addresses could be retrieved, this test case fails ("no
+   IP addresses to name servers").
+6. Create an NS query for the apex of _the child zone_.
+7. Send the NS query to each available name server IP address.
+8. If there is no response on any query this test case fails 
+   ("no response from name servers").
+9. If there is no valid response containing the NS records of 
+   _the child zone_ this test case fails 
+   ("error in response from name servers").
```

Comment:
How is the validity check to be performed?
If there are NS records in the response, how do you know if they are "the NS records of the child zone"?